zerver/views/scheduled_messages.py

```diff
@@ -23,3 +26,40 @@ def get_scheduled_message(scheduled_message_id: int, user_profile: UserProfile)
         raise ResourceNotFoundError(_("Scheduled message does not exist"))
 
 
+def fetch_scheduled_messages(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+    scheduled_messages = ScheduledMessage.objects.filter(
+        sender=user_profile, delivered=False, delivery_type=ScheduledMessage.SEND_LATER
+    )
+    scheduled_messages_to_dict: List[ScheduledMessageDict] = []
+
+    for msg in scheduled_messages:
+        to, _type = get_recipient_ids(msg.recipient, user_profile.id)
+
+        msg_to_dict: ScheduledMessageDict = {
+            "message_id": msg.id,
+            "to": to,
+            "type": _type,
+            "content": msg.content,
+            "rendered_content": msg.rendered_content,
+            "topic": msg.topic_name(),
+            "deliver_at": int(msg.scheduled_timestamp.timestamp() * 1000),
+        }
+        scheduled_messages_to_dict.append(msg_to_dict)
+
+    return json_success(request, data={"scheduled_messages": scheduled_messages_to_dict})
+
+
+def delete_scheduled_messages(
```

Comment:
It's possible these API endpoints should still have `has_request_variables`; that might be necessary for them to return `ignored_parameters_unsupported` if invalid other parameters are passed.